frontend/tests/acceptance/pages/timeLogPage.js

```diff
@@ -9,13 +10,171 @@ module.exports = {
+    this.fillNote(log.note);
+    this.clickUpdate();
+  },
+  deleteTimeLog(type) {
+    const el_delete = `${this.elements.getTimeLogContext(type)}${this.elements.deletetimelog_button}`;
+    I.waitForElement(el_delete, 5);
+    I.click(el_delete);
+  },
+  addToCalendar(type) {
+    const el_addtocalendar = `${this.elements.getTimeLogContext(type)}${this.elements.addtocalendar_button}`;
+    I.waitForElement(el_addtocalendar, 5);
+    I.click(el_addtocalendar);
+  },
+  selectType(type) {
+    if (type) {
+      I.waitForElement(this.fields.type, 5);
+      I.click(this.fields.type);
+      I.click(this.elements.select_options + '[contains(.,"' + type + '")]');
+      I.dontSeeElement(this.elements.select_options);
+    }
+  },
+  selectEventTime(calendar) {
+    I.click(this.fields.time);
+    I.waitForElement(this.datePicker.datePicker_container, 5);
+    this.selectYear(calendar.year);
+    this.selectMonth(calendar.month);
+    this.selectDay(calendar.day);
+    this.selectHour(calendar.hour);
+    this.selectMinute(calendar.minute);
+    this.selectPeriod(calendar.period);
+    I.click(this.elements.timelog_dialog);
+  },
+  selectYear(year) {
+    I.waitForElement(this.datePicker.open_year, 5);
+    I.click(this.datePicker.open_year);
+    const el_year = this.datePicker.getYear(year);
+    I.waitForElement(el_year, 5);
+    I.scrollTo(el_year);
+    I.click(el_year);
+  },
+  async selectMonth(month) {
+    I.waitForElement(this.datePicker.open_day_month, 5);
+    I.click(this.datePicker.open_day_month);
+
+    const selected = await I.waitForFunction(
+      function recurse(month, monthNames) {
+        const m_y = '//div[contains(@class,"MuiPickersSlideTransition-transitionContainer")]/p';
+        const next = '//div[contains(@class,"MuiPickersCalendarHeader-switchHeader")]/button[2]';
+        const previous = '//div[contains(@class,"MuiPickersCalendarHeader-switchHeader")]/button[1]';
+        const el_m = document.evaluate(m_y, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
+        const el_next = document.evaluate(next, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null)
+          .singleNodeValue;
+        const el_previous = document.evaluate(previous, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null)
+          .singleNodeValue;
+        let cur_month = el_m.innerText.split(' ')[0];
+        if (cur_month === month) {
+          return true;
+        } else {
+          if (monthNames.indexOf(cur_month) < monthNames.indexOf(month)) {
+            el_next.click();
+            recurse(month, monthNames);
+          } else if (monthNames.indexOf(cur_month) > monthNames.indexOf(month)) {
+            el_previous.click();
+            recurse(month, monthNames);
+          }
+        }
+      },
+      [month, monthNames],
+      100
+    );
+  },
+  async selectDay(day) {
+    const el_day = this.datePicker.getDay(day);
+    I.waitForElement(el_day, 5);
+    I.click(el_day);
```

Comment:
`el_day`, `el_pachino` are you into spanish?

Reply:
😁
wanted to mean `element_day` not `the_day`